docs/pbc/all/payment-service-provider/202404.0/base-shop/third-party-integrations/stripe/project-guidelines-for-stripe.md

```diff
@@ -80,13 +83,342 @@ This example demonstrates how to configure the order state machine transition fr
 
 ## Checkout payment step
 
+There're multiple ways to implement Stripe. Some of the options:
+
+- Add Stripe to a headless frontend application using Glue API.
+- Add it as a hosted payment page which redirect the customer after submitting an order.
+
+
+<!--- You can use the default implementation in the Payment selection page using Yves which shows the Payment Elements then later on the summary page.) -->
+
+
+### Implement Stripe in a headless application
+
+Use this approach for headless applications with third-party frontends.
+
+#### Install modules
+
+Install or upgrade the modules to the specified or higher versions:
+- `spryker/kernel-app:1.2.0`
+- `spryker/payment:5.24.0`
+- `spryker/payments-rest-api:1.3.0`
+
+#### PreOrder payment flow
+
+1. The customer either selects Stripe as the payment method or Stripe Elements is loaded by default.
+2. When Stripe is selected, the `InitializePreOrderPayment` Glue API endpoint (`glue.mysprykershop.com/payments`) is called with the following parameters:
+  * Payment provider name: Stripe
+  * Payment method name: Stripe
+  * Payment amount
+  * Quote data
+3. Zed makes the API call to the Stripe App, including required authorization.
+4. The payment with the given data is persisted On the Stripe App side.
+5. An API call to Stripe is made to get `ClientSecret` and `PublishableKey` keys.
+5. Stripe returns a JSON response with the following parameters:
+  * TransactionId
+  * ClientSecret
+  * PublishableKey
+  * Only for marketplaces: AccountId
+6. Stripe Elements is rendered on the order summary page. See [Example](#example) for rendering Stripe Elements.
+6. The customer selects a payment method in Stripe Elements and submits the data.
+7. The customer is redirected to the provided `return_url`, which makes a Glue API request to persist the order in the Back Office: `glue.mysprykershop.com/checkout`.
+8. The customer is redirected to the success page of your application.
+9. Through the `\Spryker\Zed\Payment\Communication\Plugin\Checkout\PaymentConfirmPreOrderPaymentCheckoutPostSavePlugin` plugin, the PreOrder payment is confirmed on the Stripe App side. The `order_reference` is passed to the Stripe app to be connected with `transaction_id`.
+10. After the payment is processed on the Stripe App side, a `PaymentUpdated` message is sent to Spryker.
+  The message contains additional data, which you can see in the Back Office.
+11. Depending on payment status, one of the following messages is returned through an asynchronous API, which moves the order to the next state in OMS:
+  *  When the payment is successful, a `PaymentAuthorized` message is returned.
+  *  When the payment is failed, a `PaymentAuthorizationFailed` message is returned.
+12. Further on, the order is processed through the OMS.
+
+
```

Comment:
From this point on, the information needs to be restructured based on what exactly we are trying to explain

Reply:
Is it better now?

Reply:
Nope, I still don't understand the idea behind this section. Perhaps break it down into smaller sections

Reply:
The above section describes the PreOrder Payment flow and the below sections describe where to find the example integration as well as the examples with description.

I don't see how to change it to be better.